src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -710,58 +695,12 @@ private bool IsSuspended()
         {
             return _initializing || DesignMode;
         }
-    }
-
-    internal sealed class NormalizedFilterCollection : Collection<string>
-    {
-        public NormalizedFilterCollection()
-        {
-        }
-
-        public NormalizedFilterCollection(IList<string> list) : base(list)
-        {
-        }
-
-        public void AddCollection(Collection<string> collection)
-        {
-            foreach (string item in collection)
-            {
-                this.Add(item);
-            }
-        }
-
-        protected override void InsertItem(int index, string item)
-        {
-            // Skip the string compare for "*" since it has no case-insensitive representation that differs from
-            // the case-sensitive representation.
-            base.InsertItem(index, string.IsNullOrEmpty(item) || item == "*.*" ? "*" : item);
-        }
-
-        protected override void SetItem(int index, string item)
-        {
-            if (string.IsNullOrEmpty(item))
-            {
-                // Skip the string compare for "*" since it has no case-insensitive representation that differs from
-                // the case-sensitive representation.
-                base.SetItem(index, "*");
-            }
-            else if (!string.Equals(this[index], item, PathInternal.StringComparison))
-            {
-                base.SetItem(index, item == "*.*" ? "*" : item);
-            }
-        }
 
-        protected override void ClearItems()
+        private class NormalizedFilterCollection : Collection<string>
         {
-            string filter = this[0];
-            base.ClearItems();
-            base.Add(filter);
-        }
+            protected override void InsertItem(int index, string item) => base.InsertItem(index, string.IsNullOrEmpty(item) || item == "*.*" ? "*" : item);
 
```

Comment:
Why are you bothering to add something to the list if I pass null or empty..you can leave list empty or throw

Reply:
i will leave it empty in  null or empty case